File: app/src/main/java/com/example/words/WordRepository.java
package com.example.words;

import android.content.Context;

import androidx.lifecycle.LiveData;

import java.util.List;

public class WordRepository {
    private final LiveData<List<Word>> allWordsLive;
    private final WordDao wordDao;

    public WordRepository(Context context) {
        WordDatabase wordDatabase = WordDatabase.getDatabase(context.getApplicationContext());
        wordDao = wordDatabase.getWordDao();
        allWordsLive = wordDao.getAllWordsLive();
    }

    void insertWords(Word... words) {
        wordDao.insertWords(words);
    }

    void updateWords(Word... words) {
        wordDao.updateWords(words);
    }

    void deleteWords(Word... words) {
        wordDao.deleteWords(words);
    }

    void deleteAllWords() {
        wordDao.deleteAllWords();
    }

    LiveData<List<Word>> getAllWordsLive() {
        return allWordsLive;
    }

    LiveData<List<Word>> findWordsWithPatten(String patten) {
        return wordDao.findWordsWithPatten("%" + patten + "%");
    }
}

File: app/src/main/java/com/example/words/WordRepository.kt
package com.example.words

import android.content.Context
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.flowOn
import kotlinx.coroutines.withContext

class WordRepository(context: Context) {
    private val wordDatabase by lazy { WordDatabase.getDatabase(context.applicationContext) }
    private val wordDao by lazy { wordDatabase.getWordDao() }

    suspend fun insertWords(vararg words: Word) = withContext(Dispatchers.IO) { wordDao.insertWords(*words) }

    suspend fun updateWords(vararg words: Word) = withContext(Dispatchers.IO) { wordDao.updateWords(*words) }

    suspend fun deleteWords(vararg words: Word) = withContext(Dispatchers.IO) { wordDao.deleteWords(*words) }

    suspend fun deleteAllWords() = withContext(Dispatchers.IO) { wordDao.deleteAllWords() }

    fun getAllWordsLive() = wordDao.getAllWordsLive().flowOn(Dispatchers.IO)

    fun findWordsWithPattern(pattern: String) = wordDao.findWordsWithPattern("%$pattern%").flowOn(Dispatchers.IO)
}